test_suite/src/code.rs

```diff
@@ -28,69 +29,154 @@ pub struct Code {
     noir_code: String,
     /// Input size of provided to the main function in the Noir project.
     input_size: usize,
-    /// Test case added to the main file of the noir project.
-    test_case: Option<String>,
 }
 
 impl Code {
     /// Creates a new code from the inputs of the database.
     pub fn new(regex_input: &DbEntry) -> anyhow::Result<Self> {
         let noir_code = generate_noir_code(
             &regex_input.regex,
+            regex_input.gen_substrs,
             Path::new(constants::DEFAULT_GENERATION_PATH),
         )
         .context("error generating the noir code")?;
         Ok(Self {
             noir_code,
             input_size: regex_input.input_size,
-            test_case: None,
         })
     }
 
-    /// Modifies the test case.
-    pub fn set_test_case(&mut self, test_case: &str) {
-        self.test_case = Some(String::from(test_case));
-    }
-
     /// Writes the current source code into a file in a given path.
     pub fn write_to_path(&self, path: &Path) -> anyhow::Result<()> {
-        fs::write(path, self.to_string())
+        fs::write(path, self.print_code(None, None, false))
             .context(format!("error writing the code to the path {:?}", path))?;
         Ok(())
     }
-}
+    pub fn write_test_to_path(
```

Comment:
how it worked before was that the test was "set" in the `Code` struct and then based on that the code was printed. Now with the different possibilities of testing (standard test, substring test, should fail or not) that approach would require setting *and* resetting, otherwise you're not working with a clean slate for the next test. So it is cleaner to just pass it as an argument instead of changing the properties of the struct.